fairscale/utils/testing.py

```diff
@@ -548,3 +548,14 @@ def rank(self) -> int:
 
     def size(self) -> int:
         return self._size
+
+
+class SGDWithPausingCompute(torch.optim.SGD):
+    def step(self, closure: Optional[Any] = None) -> Any:
+        loss = super().step(closure=closure)
+
+        # Add a long cuda wait on the default compute stream
+        # This is used to make sure that OSS and ShardedDDP enforce a proper stream synchronization
+        torch.cuda._sleep(1000000)
```

Comment:
so this does not work, because no real computation depends on that, so the gpu is waiting on one stream while broadcasting no the other. I need to find something a little better than that